README.md

```diff
@@ -1,3 +1,151 @@
\ No newline at end of file
+<div>
+  <img src="https://img.shields.io/badge/Node.js-v18-339933?style=for-the-badge&logo=node.js" />
+  <img src="https://img.shields.io/badge/firebase-ffca28?style=for-the-badge&logo=firebase&logoColor=black"/>
+  <img src="https://img.shields.io/badge/Angular-17-DD0031?style=for-the-badge&logo=angular&logoColor=white"/>
+  <img src="https://img.shields.io/badge/prettier-1A2C34?style=for-the-badge&logo=prettier&logoColor=F7BA3E" />
+  <img src="https://img.shields.io/badge/eslint-3A33D1?style=for-the-badge&logo=eslint&logoColor=white" />
+</div>
+
+This is a repository to support collaboration on using LLMs in behavioral economics experiments. e.g. library of relevant UI components and a library for calling LLMs.
+
+[Instance deployed at EPFL](./EPFL.md)
+
+## Project Structure
+
+```bash
+├── .vscode    # VSCode configuration
+├── firestore  # Firebase Firestore rules
+│
+├── docs       # Documentation
+│
+├── firestore  # Firestore rules and indexes
+│
+├── functions  # Firebase Cloud Functions
+│   ├── lib           # Build output
+│   ├── node_modules
+│   └── src           # Cloud functions source code
+│       ├── endpoints   # Cloud functions endpoints
+│       ├── seeders     # Model factories
+│       ├── utils       # Utilities
+│       ├── validation  # TypeBox validation utilities
+│       ├── app.ts      # Firebase app initialization
+│       └── index.ts    # Cloud functions entrypoint
+│
+└── webapp     # Webapp frontend source code
+    ├── node_modules
+    └── src               # Frontend source code
+        ├── app           # Angular JS 17 app & components
+        ├── assets        # Static assets
+        ├── environments  # Environment configuration
+        └── lib           # API, types & utilities
+```
+
+## Firebase
+
+This project uses Firebase as its backend. The configuration can be found in the [`.firebaserc`](./.firebaserc) and [`firebase.json`](./firebase.json) files.
+
+Install the firebase cli tools with the following commands:
+
+```bash
+npm install -g firebase-tools
+firebase login  # Login to the Google account destined to manage the Firebase project
+```
+
+### Emulators
+
+In order to run offline and for development purposes, we use Java Firebase emulators.
+
+```bash
+export JAVA_TOOL_OPTIONS="-Xmx4g"  # Set your desired max RAM (here: 4GB)
+firebase emulators:start  # Start the emulators
+```
+
+You will then be able to access the following UIs:
+
+- Authentication UI: `http://localhost:4000/auth`
+- Emulator UI: `http://localhost:4000`
+- Firestore UI: `http://localhost:4000/firestore`
+- Cloud Functions UI: `http://localhost:4000/functions`
+
+### Firestore
+
+We use Firestore as our database. The rules are located in the [`firestore.rules`](./firestore.rules) file.
+Basically, the database cannot be externally accessed, and must be interacted with through cloud functions.
```

Comment:
This is one way to do it, but often you can significantly simplify your cloud functions if you have a smart set of rules.

Reply:
I will think of a better firestore database layout. I can probably replace many (if not all) of the "fetcher" cloud functions by clever firestore rules and frontend logic. However, would you recommend doing this for mutating cloud functions ? I guess that in this particular case, it would be better to keep the business logic in cloud functions that can apply data validation in a secure way.

Reply:
It depends... ideally the server doesn't have to be updated much, and the server-side functions can be quite generic. I'll look over in more detail later. Sometimes you can get away with no server-side logic beyond some security rules, sometimes you need some server-side functions. @cjqian FYI